diesel/src/pg/expression/expression_methods.rs

```diff
@@ -1111,3 +1111,113 @@ impl InetOrCidr for Inet {}
+    /// Creates a PostgreSQL `||` operator
+    ///
+    /// This operator merges two `Jsonb` objects.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #     characters {
+    /// #         id -> Integer,
+    /// #         a_b_c -> Jsonb,
+    /// #         d_e_f -> Jsonb,
+    /// #     }
+    /// # }
+    /// #
+    /// # #[cfg(feature = "serde_json")]
+    /// # fn main() {
+    /// #     run_test();
+    /// # }
+    /// #
+    /// # fn run_test() {
+    /// #     use diesel::insert_into;
+    /// #     use self::characters::dsl::*;
+    /// #     use serde_json::from_str;
+    /// #
+    /// #     let connection = connection_no_data();
+    /// #     diesel::sql_query("CREATE TABLE characters (
+    /// #         id SERIAL PRIMARY KEY,
+    /// #         a_b_c JSONB NOT NULL,
+    /// #         d_e_f JSONB NOT NULL
+    /// #     )").execute(&connection)?;
+    /// #
+    /// #     // insert into a_b_c
+    /// #     let a_b_c_data = r#"
+    /// #         {
+    /// #             "a": 1,
+    /// #             "b": 2,
+    /// #             "c": 3,
+    /// #             "letters": [
+    /// #                 "a",
+    /// #                 "b",
+    /// #                 "c",
+    /// #             ]
+    /// #         }
+    /// #     "#;
+    /// #     diesel::insert_into(a_b_c).values(from_str(a_b_c_data));
+    /// #
+    /// #     // insert into d_e_f
+    /// #     let d_e_f_data = r#"
+    /// #         {
+    /// #             "d": 1,
+    /// #             "e": 2,
+    /// #             "f": 3,
+    /// #             "letters": [
+    /// #                 "d",
+    /// #                 "e",
+    /// #                 "f",
+    /// #             ]
+    /// #         }
+    /// #     "#;
+    /// #     diesel::insert_into(d_e_f).values(from_str(d_e_f_data));
+    /// #
+    ///       let a_b_c_json = a_b_c.load(&connection);
+    ///       let d_e_f_json = d_e_f.load(&connection);
+    ///
+    ///       let projected_merged_json = from_str(r#"
+    ///           {
+    ///               "a": 1,
+    ///               "b": 2,
+    ///               "c": 3,
+    ///               "d": 1,
+    ///               "e": 2,
+    ///               "f": 3,
+    ///               "letters": [
+    ///                   "a",
+    ///                   "b",
+    ///                   "c",
+    ///                   "d",
+    ///                   "e",
+    ///                   "f",
+    ///               ]
+    ///           }
+    ///       "#
+    ///       );
+    ///
+    ///       let merged_json = a_b_c_json.merge(d_e_f_json);
+    ///       assert_eq!(projected_merged_json, merged_json);
+    /// #
+    /// # }
+    /// #
+    /// # #[cfg(not(feature = "serde_json"))]
+    /// # fn main() {}
+    /// ```
+    fn merge<T: AsExpression<Jsonb>>(self, other: T) -> dsl::JsonbMerge<Self, T::Expression> {
```

Comment:
To allow nullable types here this must be changed to 
```suggestion
    fn merge<T: AsExpression<Self::SqlType>>(self, other: T) -> dsl::JsonbMerge<Self, T::Expression> {
```